# Livro para consulta:
- https://jakevdp.github.io/PythonDataScienceHandbook/03.08-aggregation-and-grouping.html
- https://jakevdp.github.io/PythonDataScienceHandbook/03.09-pivot-tables.html
    

# 1. Importando bibliotecas <a name="import"></a>

<div style="text-align: right"
     
[Voltar ao índice](#Contents)

In [1]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

%matplotlib inline

# 2. Carregando o dataframe SINASC <a name="read"></a>
<div style="text-align: right"
     
[Voltar ao índice](#Contents)

In [2]:
sinasc_raw = pd.read_csv('/kaggle/input/dados-sinasc/dados_SINASC.csv')
sinasc_raw.head()

ORIGEM   CODESTAB  CODMUNNASC  LOCNASC  IDADEMAE  ESTCIVMAE  \
0       1  2679477.0      110001        1        19        5.0   
1       1  2679477.0      110001        1        29        2.0   
2       1  2679477.0      110001        1        37        9.0   
3       1  2516500.0      110001        1        30        5.0   
4       1  2516500.0      110001        1        30        2.0   

            ESCMAE  CODOCUPMAE  QTDFILVIVO  QTDFILMORT  ...  KOTELCHUCK  \
0      8 a 11 anos         NaN         0.0         0.0  ...           5   
1      8 a 11 anos    999992.0         1.0         0.0  ...           5   
2      8 a 11 anos    513205.0         2.0         0.0  ...           5   
3  12 anos ou mais    231205.0         0.0         0.0  ...           4   
4      8 a 11 anos    999992.0         1.0         0.0  ...           5   

  CONTADOR munResStatus munResTipo               munResNome  munResUf  \
0        1        ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia   
1        2        ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia   
2        3        ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia   
3        4        ATIVO      MUNIC  Alto Alegre dos Parecis  Rondônia   
4        5        ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia   

   munResLat munResLon  munResAlt  munResArea  
0  -11.93554 -61.99982      338.0    7067.025  
1  -11.93554 -61.99982      338.0    7067.025  
2  -11.93554 -61.99982      338.0    7067.025  
3  -12.13178 -61.85308      397.0    3958.273  
4  -11.93554 -61.99982      338.0    7067.025  

[5 rows x 69 columns]

# Tarefa 1

### 1. Idade media das mães e dos pais por município (coluna munResNome)


In [3]:
sinasc_1 = sinasc_raw[['munResNome','IDADEMAE','IDADEPAI']] #criando o data frame com a idade dos pais e o municipio
sinasc_1

munResNome  IDADEMAE  IDADEPAI
0        Alta Floresta D'Oeste        19      26.0
1        Alta Floresta D'Oeste        29      24.0
2        Alta Floresta D'Oeste        37      32.0
3      Alto Alegre dos Parecis        30      24.0
4        Alta Floresta D'Oeste        30      27.0
...                        ...       ...       ...
27023                  Vilhena        32      35.0
27024              Chupinguaia        19       NaN
27025                  Vilhena        24      30.0
27026                  Vilhena        21      24.0
27027  Novo Horizonte do Oeste        18       NaN

[27028 rows x 3 columns]

In [6]:
sinasc = sinasc_1.groupby('munResNome') #atribui a distribuiçao por municipios a variabel sinasc
sinasc.mean() #calcula a media das colunas IDADEMAE e IDADEPAI por municipio.

IDADEMAE   IDADEPAI
munResNome                                     
Alta Floresta D'Oeste      25.991826  29.548701
Alto Alegre dos Parecis    24.844156  29.179104
Alto Paraíso               24.960177  28.833333
Alvorada D'Oeste           25.770732  30.757282
Ariquemes                  25.607866  32.466667
Buritis                    25.559177  30.812500
Cabixi                     26.037500  34.333333
Cacaulândia                25.546667  36.333333
Cacoal                     26.890102  30.931854
Campo Novo de Rondônia     24.825301  30.500000
Candeias do Jamari         25.243323  29.894737
Castanheiras               27.281250  30.392857
Cerejeiras                 27.240506  31.652778
Chupinguaia                25.333333  29.629630
Colorado do Oeste          27.602804  34.354839
Corumbiara                 24.716535  32.470588
Costa Marques              24.412017  30.613636
Cujubim                    24.482927  31.400000
Espigão D'Oeste            26.093333  31.198113
Governador Jorge Teixeira  24.401786  29.222222
Guajará-Mirim              25.462228  30.470588
Itapuã do Oeste            26.064220  42.500000
Jaru                       25.982974  32.197183
Ji-Paraná                  26.361137  31.132239
Machadinho D'Oeste         25.480000  31.326923
Ministro Andreazza         27.904762  33.849315
Mirante da Serra           26.304878  32.154762
Monte Negro                25.648241  34.678571
Município ignorado - RO    24.000000  22.000000
Nova Brasilândia D'Oeste   26.222222  30.520833
Nova Mamoré                25.326389  31.692308
Nova União                 26.894231  31.371429
Novo Horizonte do Oeste    26.938776  31.588235
Ouro Preto do Oeste        26.375594  31.233244
Parecis                    26.454545  32.473684
Pimenta Bueno              25.199675  29.745174
Pimenteiras do Oeste       25.075000  35.600000
Porto Velho                26.312670  32.352679
Presidente Médici          25.796053  30.150754
Primavera de Rondônia      27.023256  31.888889
Rio Crespo                 26.660000  28.000000
Rolim de Moura             26.600863  30.360479
Santa Luzia D'Oeste        26.121212  30.703297
Seringueiras               24.910714  30.500000
São Felipe D'Oeste         26.481481  31.395833
São Francisco do Guaporé   25.660066  31.638889
São Miguel do Guaporé      25.926346  32.666667
Teixeirópolis              26.031250  31.500000
Theobroma                  25.536082  30.083333
Urupá                      26.688235  31.008929
Vale do Anari              26.096774  34.000000
Vale do Paraíso            25.679012  29.976744
Vilhena                    26.433962  34.152941

### 2. Peso médio dos bebes por sexo que nasceram no dia do seu aniversário por faixas de escolaridade mae
Ex: Você, aluna(o), nasceu no dia 10/01, então você precisa filtrar o conjunto de dados nessa data e calcular o peso médio dos bebês de cada sexo por faixa de escolaridade da mãe.

In [8]:
#separando o dataframe com as variaveis SEXO,PESO, ESCMAE e DTNASC.
sinasc_2 = sinasc_raw[['SEXO', 'ESCMAE','DTNASC','PESO']]
sinasc_2

SEXO           ESCMAE      DTNASC  PESO
0      Masculino      8 a 11 anos  2019-02-19  3685
1       Feminino      8 a 11 anos  2019-02-21  3055
2       Feminino      8 a 11 anos  2019-02-25  3460
3       Feminino  12 anos ou mais  2019-03-20  2982
4       Feminino      8 a 11 anos  2019-03-23  3310
...          ...              ...         ...   ...
27023  Masculino  12 anos ou mais  2019-12-13  2110
27024  Masculino      8 a 11 anos  2019-10-04  3310
27025  Masculino      8 a 11 anos  2019-08-02  3065
27026  Masculino      8 a 11 anos  2019-12-23  2352
27027  Masculino      8 a 11 anos  2019-12-27  4240

[27028 rows x 4 columns]

In [17]:
niver = sinasc_2[sinasc_2['DTNASC']=='2019-01-31'] #filtra pela minha data de nascimento 31/01

In [21]:
peso_med = niver.groupby(['ESCMAE','SEXO'])['PESO'].mean().unstack() #usa o filtro anterior e agrupa por escolaridade e sexo, calc a media dos pesos
peso_med

SEXO                Feminino    Masculino
ESCMAE                                   
1 a 3 anos       3205.000000          NaN
12 anos ou mais  2974.166667  3152.000000
4 a 7 anos       3400.000000  3020.428571
8 a 11 anos      3377.680000  3373.333333

### 3. Qual o municipio que nasceu menos bebe em 2019?
    - qual a idade media, maxima, minima das maes nesse municipio?
    - qual a idade media, maxima, minima dos pais nesse municipio?

In [23]:
#separando o dataframe com as variaveis a serem utilizadas: IDADEMAE, IDADEPAI,munResNome.
sinasc_3 = sinasc_raw[[ 'munResNome','IDADEMAE', 'IDADEPAI']]
sinasc_3

munResNome  IDADEMAE  IDADEPAI
0        Alta Floresta D'Oeste        19      26.0
1        Alta Floresta D'Oeste        29      24.0
2        Alta Floresta D'Oeste        37      32.0
3      Alto Alegre dos Parecis        30      24.0
4        Alta Floresta D'Oeste        30      27.0
...                        ...       ...       ...
27023                  Vilhena        32      35.0
27024              Chupinguaia        19       NaN
27025                  Vilhena        24      30.0
27026                  Vilhena        21      24.0
27027  Novo Horizonte do Oeste        18       NaN

[27028 rows x 3 columns]

In [30]:
#municipio que menos nasceu bebes
sinasc_3['munResNome'].value_counts().sort_values()

munResNome
Município ignorado - RO         1
Castanheiras                   32
Pimenteiras do Oeste           40
Primavera de Rondônia          43
Parecis                        44
Rio Crespo                     50
São Felipe D'Oeste             54
Teixeirópolis                  64
Cacaulândia                    75
Cabixi                         80
Vale do Paraíso                81
Ministro Andreazza             84
Vale do Anari                  93
Theobroma                      97
Novo Horizonte do Oeste        98
Santa Luzia D'Oeste            99
Nova União                    104
Itapuã do Oeste               109
Governador Jorge Teixeira     112
Corumbiara                    127
Alto Alegre dos Parecis       154
Chupinguaia                   162
Mirante da Serra              164
Campo Novo de Rondônia        166
Seringueiras                  168
Urupá                         170
Monte Negro                   199
Alvorada D'Oeste              205
Cujubim                       205
Col

In [ ]:
# O municipio com menos nascimento foi "Município ignorado - RO"
# acredito ser apenas uma informaçao generica, entao o municipio com menor quantidade de nacismentos é "Castanheiras".

In [29]:
#agrupando os dados por municipio e agragando as idades max,med e min das maes e pais
sinasc_3.dropna().groupby('munResNome').agg([('mínimo','min'),('média','mean'),('máximo', 'max')])

IDADEMAE                   IDADEPAI             \
                            mínimo      média máximo   mínimo      média   
munResNome                                                                 
Alta Floresta D'Oeste           14  25.892857     53     17.0  29.548701   
Alto Alegre dos Parecis         14  24.679104     41     16.0  29.179104   
Alto Paraíso                    16  22.833333     32     21.0  28.833333   
Alvorada D'Oeste                15  26.825243     40     17.0  30.757282   
Ariquemes                       15  28.133333     39     20.0  32.466667   
Buritis                         16  24.750000     42     18.0  30.812500   
Cabixi                          18  32.166667     38     19.0  34.333333   
Cacaulândia                     18  31.000000     42     28.0  36.333333   
Cacoal                          14  26.978605     47     17.0  30.931854   
Campo Novo de Rondônia          17  24.900000     40     21.0  30.500000   
Candeias do Jamari              15  23.684211     34     21.0  29.894737   
Castanheiras                    18  28.071429     39     17.0  30.392857   
Cerejeiras                      15  28.361111     42     19.0  31.652778   
Chupinguaia                     17  26.888889     38     20.0  29.629630   
Colorado do Oeste               23  31.161290     41     23.0  34.354839   
Corumbiara                      17  29.000000     43     21.0  32.470588   
Costa Marques                   17  25.250000     41     17.0  30.613636   
Cujubim                         17  27.400000     37     19.0  31.400000   
Espigão D'Oeste                 14  26.808176     44     16.0  31.198113   
Governador Jorge Teixeira       17  25.555556     34     22.0  29.222222   
Guajará-Mirim                   19  26.058824     39     19.0  30.470588   
Itapuã do Oeste                 27  31.375000     39     25.0  42.500000   
Jaru                            15  28.450704     44     19.0  32.197183   
Ji-Paraná                       13  27.282819     44     15.0  31.132239   
Machadinho D'Oeste              13  26.048077     40     17.0  31.326923   
Ministro Andreazza              17  27.890411     44     21.0  33.849315   
Mirante da Serra                16  26.928571     41     20.0  32.154762   
Monte Negro                     15  27.642857     40     20.0  34.678571   
Município ignorado - RO         24  24.000000     24     22.0  22.000000   
Nova Brasilândia D'Oeste        15  27.104167     38     20.0  30.520833   
Nova Mamoré                     15  26.051282     46     19.0  31.692308   
Nova União                      16  27.371429     39     18.0  31.371429   
Novo Horizonte do Oeste         16  26.988235     41     20.0  31.588235   
Ouro Preto do Oeste             14  27.109920     44     17.0  31.233244   
Parecis                         17  26.973684     41     18.0  32.473684   
Pimenta Bueno                   13  25.449807     47     16.0  29.745174   
Pimenteiras do Oeste            27  32.600000     40     25.0  35.600000   
Porto Velho                     12  28.355655     44     16.0  32.352679   
Presidente Médici               15  26.170854     43     16.0  30.150754   
Primavera de Rondônia           17  28.083333     39     21.0  31.888889   
Rio Crespo                      26  26.000000     26     28.0  28.000000   
Rolim de Moura                  14  26.549701     44     16.0  30.360479   
Santa Luzia D'Oeste             15  26.043956     41     18.0  30.703297   
Seringueiras                    15  25.058824     39     20.0  30.500000   
São Felipe D'Oeste              17  26.979167     41     18.0  31.395833   
São Francisco do Guaporé        15  26.944444     40     19.0  31.638889   
São Miguel do Guaporé           17  27.469136     42     19.0  32.666667   
Teixeirópolis                   16  27.454545     39     20.0  31.500000   
Theobroma                       19  24.916667     38     23.0  30.083333   
Urupá                           15  27.116071     43     17.0  31.008929  

### 4. Qual o municipio que nasceu mais bebe no mês de março?
    - qual a quantidade de filhos vivos media, maxima, minima nesse municipio?
    - qual a idade media, maxima, minima dos pais nesse municipio?



In [36]:
#df
sinasc_4 = sinasc_raw[['munResNome','DTNASC','QTDFILVIVO','IDADEMAE','IDADEPAI']].dropna()
sinasc_4

munResNome      DTNASC  QTDFILVIVO  IDADEMAE  IDADEPAI
0        Alta Floresta D'Oeste  2019-02-19         0.0        19      26.0
1        Alta Floresta D'Oeste  2019-02-21         1.0        29      24.0
2        Alta Floresta D'Oeste  2019-02-25         2.0        37      32.0
3      Alto Alegre dos Parecis  2019-03-20         0.0        30      24.0
4        Alta Floresta D'Oeste  2019-03-23         1.0        30      27.0
...                        ...         ...         ...       ...       ...
27021        Colorado do Oeste  2019-11-20         1.0        25      38.0
27022                  Vilhena  2019-11-21         0.0        34      33.0
27023                  Vilhena  2019-12-13         1.0        32      35.0
27025                  Vilhena  2019-08-02         0.0        24      30.0
27026                  Vilhena  2019-12-23         1.0        21      24.0

[7450 rows x 5 columns]

In [46]:
#municipios que nasceram mais bebes no mes de março

bebes_mar = sinasc_4['DTNASC'] = pd.to_datetime(sinasc_4['DTNASC'])
bebes_mar = bebes_mar.dt.month == 3
bb_mar = sinasc_4[bebes_mar]

In [80]:

mun_mais_nasc = bb_mar['munResNome'].value_counts().idxmax()
mun_mais_nasc

'Cacoal'

In [81]:
count_bb = bb_mar['munResNome'].value_counts().sort_values(ascending = False)
count_bb

munResNome
Cacoal                      122
Ji-Paraná                    95
Rolim de Moura               66
Pimenta Bueno                54
Porto Velho                  48
Ouro Preto do Oeste          40
Espigão D'Oeste              30
Alta Floresta D'Oeste        24
Vilhena                      16
Buritis                      15
Presidente Médici            14
Machadinho D'Oeste           13
Urupá                        13
Alvorada D'Oeste             10
Alto Alegre dos Parecis       9
Nova Mamoré                   9
Jaru                          9
Cerejeiras                    9
Santa Luzia D'Oeste           7
Ministro Andreazza            6
Vale do Paraíso               6
Monte Negro                   6
Novo Horizonte do Oeste       5
Chupinguaia                   5
Parecis                       5
Castanheiras                  4
Primavera de Rondônia         4
São Felipe D'Oeste            4
Colorado do Oeste             4
Nova Brasilândia D'Oeste      3
Nova União                   

In [72]:
print('O municipio com maior quantidade de nascimentos no mes de março foi:', mun_mais_nasc,'com',count_bb.iloc[0], 'nascimentos.')

O municipio com maior quantidade de nascimentos no mes de março foi: Cacoal com 122 nascimentos.


In [62]:
bb_mar.dtypes

munResNome            object
DTNASC        datetime64[ns]
QTDFILVIVO           float64
IDADEMAE               int64
IDADEPAI             float64
dtype: object

In [65]:
#calculando a idade media dos pais e a quantidade de filhos vivos nesse municipio
bb_mar[bb_mar['munResNome'] == 'Cacoal'].groupby(['munResNome', 'DTNASC']).agg([('Minimo',min),('Media','mean'),('Maximo',max)])

/tmp/ipykernel_33/162845152.py:2: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "min" instead.
  bb_mar[bb_mar['munResNome'] == 'Cacoal'].groupby(['munResNome', 'DTNASC']).agg([('Minimo',min),('Media','mean'),('Maximo',max)])
/tmp/ipykernel_33/162845152.py:2: FutureWarning: The provided callable <built-in function max> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  bb_mar[bb_mar['munResNome'] == 'Cacoal'].groupby(['munResNome', 'DTNASC']).agg([('Minimo',min),('Media','mean'),('Maximo',max)])
/tmp/ipykernel_33/162845152.py:2: FutureWarning: The provided callable <built-in function min> is currently using SeriesGroupBy.min. In a future version of pandas, the provided callable will be used directly.

QTDFILVIVO                  IDADEMAE                    \
                          Minimo     Media Maximo   Minimo      Media Maximo   
munResNome DTNASC                                                              
Cacoal     2019-03-01        1.0  1.000000    1.0       25  26.500000     28   
           2019-03-02        0.0  0.571429    2.0       18  24.714286     39   
           2019-03-03        3.0  3.000000    3.0       27  27.000000     27   
           2019-03-04        0.0  0.428571    1.0       20  27.000000     32   
           2019-03-05        0.0  3.000000    5.0       30  34.000000     38   
           2019-03-06        0.0  0.500000    1.0       16  21.500000     27   
           2019-03-07        0.0  0.333333    1.0       18  23.166667     28   
           2019-03-08        0.0  1.400000    3.0       18  29.600000     40   
           2019-03-09        0.0  0.666667    1.0       24  31.000000     39   
           2019-03-10        1.0  2.000000    3.0       26  26.000000     26   
           2019-03-11        0.0  1.800000    3.0       22  27.200000     34   
           2019-03-12        0.0  1.000000    2.0       18  24.000000     31   
           2019-03-13        1.0  1.000000    1.0       31  31.000000     31   
           2019-03-14        0.0  1.300000    3.0       21  27.700000     36   
           2019-03-15        0.0  1.400000    3.0       20  29.000000     39   
           2019-03-16        0.0  0.833333    2.0       15  26.666667     33   
           2019-03-17        0.0  0.500000    1.0       17  21.000000     26   
           2019-03-18        0.0  0.714286    3.0       18  25.571429     33   
           2019-03-19        0.0  0.600000    2.0       23  30.000000     36   
           2019-03-20        0.0  3.500000    7.0       28  31.000000     34   
           2019-03-21        0.0  0.000000    0.0       18  22.750000     31   
           2019-03-22        0.0  0.800000    2.0       25  28.400000     32   
           2019-03-23        1.0  1.000000    1.0       24  24.000000     24   
           2019-03-24        1.0  1.000000    1.0       23  23.000000     23   
           2019-03-25        0.0  0.600000    1.0       18  25.400000     35   
           2019-03-27        0.0  0.000000    0.0       22  26.000000     30   
           2019-03-28        0.0  0.800000    1.0       21  29.000000     36   
           2019-03-29        0.0  0.666667    1.0       19  25.000000     31   
           2019-03-30        0.0  0.666667    2.0       17  23.666667     29   
           2019-03-31        0.0  0.000000    0.0       19  19.000000     19   

                      IDADEPAI                    
                        Minimo      Media Maximo  
munResNome DTNASC                                 
Cacoal     2019-03-01     29.0  36.500000   44.0  
           2019-03-02     17.0  28.857143   38.0  
           2019-03-03     25.0  25.000000   25.0  
           2019-03-04     17.0  29.428571   41.0  
           2019-03-05     30.0  33.333333   37.0  
           2019-03-06     21.0  26.750000   30.0  
           2019-03-07     22.0  28.666667   47.0  
           2019-03-08     25.0  33.200000   44.0  
           2019-03-09     23.0  31.666667   36.0  
           2019-03-10     25.0  29.500000   34.0  
           2019-03-11     21.0  28.000000   33.0  
           2019-03-12     22.0  27.500000   31.0  
           2019-03-13     32.0  32.000000   32.0  
           2019-03-14     25.0  32.400000   43.0  
           2019-03-15     25.0  31.200000   43.0  
           2019-03-16     18.0  28.500000   35.0  
           2019-03-17     24.0  29.500000   38.0  
           2019-03-18     19.0  27.571429   34.0  
           2019-03-19     30.0  34.800000   39.0  
           2019-03-20     31.0  35.500000   40.0  
           2019-03-21     20.0  25.250000   35.0  
           2019-03-22     20.0  32.800000   46.0  
           2019-03-23     33.0  33.000000   33.0  
           2019-03-24     39.0  39.000000   39.0  
    

### Analise as respostas encontradas, tire algum insight delas, conte pra gente algo encontrado nos dados. Algo que você julgue relevante e novo pra você.

Exemplo:
- Ah, descobri que a idade mediana das mulheres que deram a luz no ano de 2019 dos municipios x é maior que y.

Umas das coisas que eu percebi fazendo as analises nessa base de dados, é o quao relevante pode ser a falta de dados, ou seja, os missings. Como para calcular os valores das idades dos pais, foi removido as linhas com os dados faltantes. Isso mudou drasticamente o resultado em relaçao ao municipio com mais nascimentos em março. Na atividade proposta eu removi os dados faltantes resultando como municipio com maior numero de nascimentos o municipio de Cacoal. Porem refazendo a mesma consulta sem remover os dados faltantes temos:

In [ ]:
#refazendo os dados do eexercicio numero 4 sem retirar os dados faltantes.

In [73]:
sinasc_conclud = sinasc_raw[['munResNome','DTNASC','QTDFILVIVO','IDADEMAE','IDADEPAI']]

In [74]:
bebes_mar_com_na = sinasc_conclud['DTNASC'] = pd.to_datetime(sinasc_conclud['DTNASC'])
bebes_mar_com_na = bebes_mar_com_na.dt.month == 3
bb_mar_com_na = sinasc_conclud[bebes_mar_com_na]

/tmp/ipykernel_33/776607723.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  bebes_mar_com_na = sinasc_conclud['DTNASC'] = pd.to_datetime(sinasc_conclud['DTNASC'])


In [78]:
mun_mais_nasc1 = bb_mar_com_na['munResNome'].value_counts().idxmax()
mun_mais_nasc1

In [82]:
count_bb1 = bb_mar_com_na['munResNome'].value_counts().sort_values(ascending = False)
count_bb1

munResNome
Porto Velho                  744
Ji-Paraná                    188
Vilhena                      148
Ariquemes                    141
Cacoal                       133
Jaru                          80
Guajará-Mirim                 75
Rolim de Moura                69
Ouro Preto do Oeste           67
Pimenta Bueno                 62
Buritis                       59
Nova Mamoré                   53
Espigão D'Oeste               44
Machadinho D'Oeste            44
São Miguel do Guaporé         41
Candeias do Jamari            33
São Francisco do Guaporé      31
Alta Floresta D'Oeste         31
Presidente Médici             29
Costa Marques                 24
Alto Paraíso                  22
Alvorada D'Oeste              22
Monte Negro                   20
Nova Brasilândia D'Oeste      20
Seringueiras                  19
Colorado do Oeste             19
Chupinguaia                   18
Cujubim                       17
Cerejeiras                    16
Urupá                         14

Ou seja, sem remover as linhas que contiam dados faltantes referente a idade do pai, outras 4 cidades aparecem na frente de Cacoal com mais nascidos em Março, com uma quantidade bem superior. Isso nos leva a outra analise. Os pais de Cacoal em sua maioria estavam presentes para fornecer a idade, tendo uma diferença de 133 sem tirar os dados faltantes para 122 removendo os dados faltantes. Ja o municipio de Porto Velho, capital do Estado, já era esperado ter o maior numero de nascimentos porem dos 744 registros, apenas 48 registraram a idade dos pais.

In [83]:
#verificando porcentagem de dados faltantes para a idade do Pai

sinasc_raw['IDADEPAI'].isna().value_counts()

IDADEPAI
True     19421
False     7607
Name: count, dtype: int64

In [85]:
dados_faltantes_porcent = 19421/(19421+7607)
dados_faltantes_porcent

0.7185511321592423

Nesse caso que a maioria das linhas possuiem dados faltantes (cerca de 71%), por isso, não acho viavel realizar a remoção dessas linhas apenas para analisar a idade dos pais (Masculino). 